Assets/Mirror/Runtime/NetworkConnection.cs

```diff
@@ -99,18 +99,20 @@ public bool InvokeHandlerNoData(short msgType)
             return InvokeHandler(msgType, null);
         }
 
+        // InvokeHandler is in hot path. caching the NetworkMessage is really
+        // worth it to avoid large amounts of allocations.
+        static NetworkMessage invokeHandlerMsg = new NetworkMessage();
```

Comment:
This is not needed at all if NetworkMessage is a struct.    If it is a struct it does no allocation.

Reply:
that would be good. will run a stress test.

Reply:
ok, this seems to cause errors anyway:

Component [0] not found for [netId=1]
UnityEngine.Debug:LogWarning(Object)
Mirror.NetworkIdentity:HandleRemoteCall(Int32, Int32, UNetInvokeType, NetworkReader) (at Assets/uMMORPG/Plugins/Mirror/Runtime/NetworkIdentity.cs:594)
Mirror.NetworkIdentity:HandleRPC(Int32, Int32, NetworkReader) (at Assets/uMMORPG/Plugins/Mirror/Runtime/NetworkIdentity.cs:613)
Mirror.ClientScene:OnRPCMessage(NetworkMessage) (at Assets/uMMORPG/Plugins/Mirror/Runtime/ClientScene.cs:621)
Mirror.NetworkConnection:HandleBytes(Byte[]) (at Assets/uMMORPG/Plugins/Mirror/Runtime/NetworkConnection.cs:222)
Mirror.NetworkConnection:TransportReceive(Byte[]) (at Assets/uMMORPG/Plugins/Mirror/Runtime/NetworkConnection.cs:272)
Mirror.NetworkClient:OnDataReceived(Byte[]) (at Assets/uMMORPG/Plugins/Mirror/Runtime/NetworkClient.cs:98)
UnityEngine.Events.UnityEvent`1:Invoke(Byte[])
Mirror.ErlangenTransport:ProcessClientMessage() (at Assets/ErlangenTransport.cs:63)
Mirror.ErlangenTransport:LateUpdate() (at Assets/ErlangenTransport.cs:85)

removed it for now.